src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -523,6 +526,19 @@ override public void Open()
         {
             PrepareStatisticsForNewConnection();
 
+            if (StatisticsEnabled ||
```

Comment:
@saurabh500 I have refactored this logic into `PrepareStatisticsForNewConnect()` to eliminate the redundancy.  These changes can be found in commit https://github.com/dotnet/corefx/pull/8420/commits/319736ba1954415602e7aecf25ba52daee275f36 Thanks! :smile: